lib/sparql_converter.js

```diff
@@ -0,0 +1,386 @@
+"use strict";
+
+require("../test/polyfill");
+
+const query_limit = 10;
+
+//main template for wikidata query
+const q = `
+SELECT distinct ?v1 ?v1Label ?v2Label ?v3Label WHERE{
+
+%s
+%s
+%s
+%s
+%s
+
+SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
+
+}
+%s
+limit ${query_limit}
+`;
+
+//wikidata triplet templates
+
+//gets wikidata representation of raw values (e.g. San Francisco = Q62)
+const get_label_triplet = `?item%d ?label '%s'@en.\n`;
+
+//makes query based on wikidata representation
+const get_label_parameter_triplet = `?v1 wdt:%s ?item%d.\n`;
+
+//for some values(e.g. numbers) there is no wikidata represntation
+const no_label_triplet = `?v1 wdt:%s %s.\n`;
+
+//triplet required if there is a join
+const get_join_triplet = `?v2 wdt:%s ?v1.\n`;
+
+//triplet required if there is a projection
+const get_projection_triplet = `?v1 wdt:%s ?v3.\n`;
+
+//triplet required if there is a projection
+const sort_results_triplet = `?v%s wdt:%s ?counter.\n`;
+
+//for ordering results either ascending or descending
+const order_template = `ORDER BY %s(?counter)`;
+
+module.exports = {
+    toSparql(input) {
```

Comment:
This is very much Wikidata specific, and the name should reflect it.
Also, given that there is only one wikidata, I wonder what's the advantage of having this in the main thingtalk library...